x-pack/plugins/apm/public/components/shared/PropertiesTable/index.js

```diff
@@ -156,6 +53,17 @@ function AgentFeatureTipMessage({ featureName, agentName }) {
   );
 }
 
+function sortKeysByConfig(object, currentKey) {
+  const presorted = _.get(
+    indexedPropertyConfig,
+    `${currentKey}.presortedKeys`,
+    []
+  );
+  return _.uniq([...presorted, ...Object.keys(object).sort()]);
+}
+
+// TODO: if propData defaults to an empty object, the if (!propData)
+// will never be truthy unless propData is explicitly null or false...
```

Comment:
Good point. I wonder if you should just remove the default value. But then ensure that `NestedKeyValueTable` doesn't blow up when `data` is `undefined`.